### DISCUSSION TITLE
Is it possible to replace the text in the DatePickerInput button with a custom format?

### DISCUSSION THREAD

**Discussion by ian-price-g2o** [mantinedev/mantine#4714] [Q&A]
I'm trying to use the `type="multiple"` functionality of the DatePickerInput component, and was wondering if the list of dates that shows up in the rendered button can be altered. Basically I'm limited with the vertical space I can have with this component and was hoping that if more than 2 dates were selected I could replace the text with something like: `3 Dates Selected`. Is this possible?

**Comment by rtivital** [comment]
No, currently that is not supported, you can create a custom date picker with Calendar, Input and Popover components

**Reply by ian-price-g2o** [reply]
Got it, thank you.

**Comment by ian-price-g2o** [comment]
This may not be a best practice, but I've written a small useEffect snippet to check the props value and update the DOM directly when the value is an array with a length greater than 2:

```
const buttonRef = useRef(null);

useEffect(() => {
  if(props.type === "multiple" && buttonRef.current) {
    if(Array.isArray(props.value) && props.value.length > 2) {
      const el = buttonRef.current as HTMLButtonElement;
      el.innerText = `${props.value.length} Dates`;
    }
  }
},[props.value])
```
